control-plane/api/v1alpha1/servicedefaults_types.go

```diff
@@ -244,17 +247,22 @@ func (in *ServiceDefaults) SyncedConditionStatus() corev1.ConditionStatus {
 
 // ToConsul converts the entry into it's Consul equivalent struct.
 func (in *ServiceDefaults) ToConsul(datacenter string) capi.ConfigEntry {
+	maxInboundConnections := 0
+	if in.Spec.MaxInboundConnections != nil {
+		maxInboundConnections = *in.Spec.MaxInboundConnections
+	}
```

Comment:
If Consul can accept an `int`, and it does default to `0` which implies unlimited, I think we can set the type as just `int` instead of `*int`. Kube values that tend to use `*int` is because they do care that 0 isn't set accidently. This should simplify the logic here as it can be a direct assignment!!